src/Microsoft.AspNetCore.Authentication.Facebook/FacebookOptions.cs

```diff
@@ -57,6 +60,6 @@ public string AppSecret
         /// The list of fields to retrieve from the UserInformationEndpoint.
         /// https://developers.facebook.com/docs/graph-api/reference/user
         /// </summary>
-        public IList<string> Fields { get; }
+        public ICollection<string> Fields { get; } = new HashSet<string>();
```

Comment:
Let's see what @Tratcher  has to say.  I think your right though ISet<T> would give a better indication to the user and if we are happy that it will always be a hash set then that's ok..